### ISSUE TITLE
SimCity 2000 gets stuck on some dosbox window saying "exit" isn't a command

### ISSUE THREAD

**Issue by ell1e** [sharkwouter/minigalaxy#592]
I just tried launching SimCity 2000 and it gets stuck on some dosbox window saying "exit" isn't a command (during that entire process I didn't touch anything, it ran those commands by itself):

![Screenshot from 2024-01-16 00-11-10](https://github.com/sharkwouter/minigalaxy/assets/64124388/984305bb-1fef-437f-8ff3-4674b5217652)

Am I doing something wrong maybe? :flushed: I installed it via flatpak/flathub, latest version on there, and I'm testing it on an openSUSE Slowroll x64 machine with a GNOME 3 Wayland session

**Comment by ell1e** [comment]
I'm seeing the same with Lands of Lore 1: Throne of Chaos. Maybe it's a problem with all dosbox games somehow?

**Comment by sharkwouter** [comment]
Yeah, unfortunately this happens with all dosbox Windows games. I think it's a wine issue.

**Comment by ell1e** [comment]
Is there any known workaround to this? Sorry to eat up anyone's time, just curious. It would be nice if it were possible to launch those games somehow, through minigalaxy.

**Comment by sharkwouter** [comment]
Yeah, you can install dosbox on your machine. Then it should start.

**Comment by ell1e** [comment]
Dosbox is already installed!

**Comment by sharkwouter** [comment]
Oh, my bad. You might want to report this issue here, since you installed using Flatpak: https://github.com/flathub/io.github.sharkwouter.Minigalaxy/issues

**Comment by ell1e** [comment]
I reported it here now: https://github.com/flathub/io.github.sharkwouter.Minigalaxy/issues/52

**Comment by ell1e** [comment]
I found out after more experimenting that only some games are affected, which I assume suggests the problem may not be flatpak since dosbox seems to be available but only work correctly *sometimes*.

Working DOS games:

- Master of Orion (original DOS version, not the remaster)
- Pizza Connection
- Blake Stone Planet Strike
- Star Wars Dark Forces (original DOS version, not the remaster)

Broken DOS games:

- Blake Stone  Aliens of Gold
- Alone in the Dark (original DOS version, not any sequel or reboot with same name)
- Alone in the Dark 2
- Alone in the Dark 3
- Settlers 2

**Comment by ell1e** [comment]
I noticed that the working DOS games don't have a wine glass icon showing.

Is it possible that minigalaxy tries to detect DOS Games and to use dosbox on them directly, but messes that up somehow and tries to run dosbox in wine for the failed ones and that's what breaks? Although I don't fully understand why, since I've run dosbox in wine before using Steam and there it usually works perfectly fine. But it would explain the different behavior.

**Comment by sharkwouter** [comment]
Minigalaxy will use the system's Doxbox for Dosbox games that are released for Linux. For Window Dosbox games it will just use wine. Not sure if that helps.

**Comment by ell1e** [comment]
It seems like the Linux Dosbox games work then, and the Windows Dosbox games don't.

**Comment by ell1e** [comment]
Actually, I was incorrect: I just found a Windows Dosbox game that works, [Witchaven (the first one of the two combined in this bundle)](https://www.gog.com/en/game/witchaven_witchaven_ii_blood_vengeance). But it seems like at least a large amount of Windows Dosbox games have this weird `Illegal command: "exit".` problem, while none of the Linux Dosbox games I tried seem to have it.

**Comment by ell1e** [comment]
Would you be interested in me gifting you one of the affected games so you can try out if it works for you, and if not, figure out how to possibly fix it? (I understand you may still not figure that out, but at least it might raise the chances.)

**Comment by sharkwouter** [comment]
I think for this one, I have some games I could test. Thanks for the offer, though.

**Comment by sharkwouter** [comment]
This issue happened when dosbox games for Windows were just extracted using innoextract instead of being installed. This has been fixed for a while now.
